server/routes/products/searchProducts.js

```diff
@@ -55,36 +58,57 @@ router.get("/products", (req, res) => {
               `%${req.query.keyword.toLowerCase()}%`
             );
         } else {
-          builder
-            .where(
-              "products.name",
-              "ILIKE",
-              `%${req.query.keyword.toLowerCase()}%`
-            )
-            .orWhere(
-              "c.name",
-              "ILIKE",
-              `%${req.query.keyword.toLowerCase()}%`
-            );
+          if (parentCategory.length > 0) {
+            builder
+              .where(
+                "products.name",
+                "ILIKE",
+                `%${req.query.keyword.toLowerCase()}%`
```

Comment:
what if i remove the keyword from url, this will crash right? you need to handle such cases more importantly